pywemo/ssdp.py

```diff
@@ -252,15 +282,17 @@ def scan(st=None, timeout=DISCOVER_TIMEOUT, max_entries=None, match_mac=None, ma
 
                 # Search for devices
                 if st is not None or match_mac is not None or match_serial is not None:
-                    if st is not None and st == entry.st:
-                        entries.append(entry)
-
-                    if entry not in entries and match_mac is not None and match_mac == mac:
-                        entries.append(entry)
-
-                    if entry not in entries and match_serial is not None and match_serial == serial:
-                        entries.append(entry)
-                elif entry not in entries:
+                    if not entry_in_entries(entry, entries):
```

Comment:
Since entry_in_entries also turns the entry object into a device object, why not pass in the device object from line 276 to entry_in_entries and use that instead of having entry_in_entries do repetitive work to what has already been done in line 276?